lib/mercury/mercury.rb

```diff
@@ -44,10 +50,14 @@ def initialize(host: 'localhost',
   def publish(source_name, msg, tag: '', headers: {}, &k)
     # The amqp gem caches exchange objects, so it's fine to
     # redeclare the exchange every time we publish.
-    # TODO: wait for publish confirmations (@channel.on_ack)
     with_source(source_name) do |exchange|
-      exchange.publish(write(msg), **Mercury.publish_opts(tag, headers)) do
-        k.call
+      payload = write(msg)
+      pub_opts = Mercury.publish_opts(tag, headers)
+      if publisher_confirms_enabled
+        expect_publisher_confirm(k)
```

Comment:
Good question. See https://www.rabbitmq.com/blog/2012/04/25/rabbitmq-performance-measurements-part-2/  You're right, our messaging rates are far below where publisher confirms would have a noticeable impact.